File: Bitwise Operation/137. Single Number II/main.go
package main

import (
	"fmt"
)

// use the same strategy in two sum
func singleNumber(nums []int) int {
	m := make(map[int]int)
	for _, num := range nums {
		m[num]++ // count
	}
	for key, val := range m {
		if val == 1 {
			return key
		}
	}
	return 0
}

func main() {
	nums := []int{0, 1, 0, 1, 0, 1, 99}
	fmt.Println("Output: ", singleNumber(nums))
}

File: Bitwise Operation/137. Single Number II/main.c
#include <stdio.h>
#include <stdlib.h>

int singleNumber(int* nums, int numsSize){
    int a = 0, b = 0, c = 0;
    int i = 0;
	for (i = 0; i < numsSize; i++) {
		a = (b & ~nums[i]) | (c & nums[i]);
		c = (~b & ~c & nums[i]) | (c & ~nums[i]);
		b = a;
	}
	return c;
}

int main() {
    int nums[] = {0, 1, 0, 1, 0, 1, 99};
    printf("Output: %d.", singleNumber(nums, 7));
    return 0;
}